integration-tests/tests/src/tests/dictionary.ts

```diff
@@ -333,4 +431,342 @@ describe("Dictionary", () => {
     },
     expectedError: "JS value must be of type 'object'",
   });
+  describe("Dictionary queries", () => {
+    const DictSchema: Realm.ObjectSchema = {
+      name: "Dictionary",
+      properties: {
+        a: "{}",
+      },
+    };
+    let realm: Realm;
+    beforeEach(() => {
+      Realm.clearTestState();
+      realm = new Realm({ schema: [DictSchema] });
+    });
+    it("support filters", () => {
+      const N = 100;
+      for (let i = 0; i < N; i++) {
+        realm.write(() => realm.create(DictSchema.name, { a: { x: i, y: 2, z: 3 } }));
+      }
+
+      const data = realm.objects(DictSchema.name);
+      expect(data.length).equals(N, "We expect ${N} objects.");
+
+      const half = data.filtered("a['x'] >= 50");
+      const seventy = data.filtered("a['x'] >= $0", 70);
+      expect(half.length).equals(N / 2, "We expect only 50 items, matching for field x.");
+      expect(seventy.length).equals(30, "We expect only 30 items, matching for field x.");
+    });
+  });
+  describe("notifications", () => {
+    const DictSchema: Realm.ObjectSchema = {
+      name: "Dictionary",
+      properties: {
+        fields: "{}",
+      },
+    };
+    let realm;
+    beforeEach(() => {
+      Realm.clearTestState();
+      realm = new Realm({ schema: [DictSchema] });
+    });
+    it("support update notifications", () => {
+      const UPDATES = 5;
+      realm.write(() => realm.create(DictSchema.name, { fields: { field1: 0, filed2: 2, field3: 3 } }));
+      const fields = realm.objects(DictSchema.name)[0].fields;
+      let cnt = 0;
+      fields.addListener((_, changeset) => {
+        expect(fields.field1).equals(cnt, "fields.field1: ${fields.field1} should be equals to: cnt -> ${cnt}");
+        // We ignore the first as it just reflect the creation in the line above.
+        if (cnt > 0) {
+          expect(changeset.modifications[0]).equals(
+            "field1",
+            "The changeset should reflect an update on field1 but it shows -> ${changeset.modifications[0]}",
+          );
+        }
+        cnt++;
+      });
+
+      for (let i = 1; i <= UPDATES; i++) {
+        realm.write(() => {
+          fields.field1 = i;
+        });
+      }
+
+      return new Promise<void>((resolve, reject) => {
+        setTimeout(() => {
+          expect(realm.objects(DictSchema.name)[0].fields.field1).equals(5);
+          expect(cnt).equals(UPDATES + 1, "We expect ${UPDATES + 1} updates.");
+          fields.removeAllListeners();
+          realm.close();
+          resolve();
+        }, 1000);
+      });
+    });
+    // FIXME: Running the following test produce unexpected side-effects which breaks all subsequent tests.
+    // See https://github.com/realm/realm-js/issues/3834
+    it.skip("support insert notifications", () => {
```

Comment:
this wasn't used in the regular js file either but I thought it'd be worth moving over

Reply:
This works now, so you can uncomment and migrate this as well.